### PR TITLE
charliecloud: add versions 0.28 and 0.29

### PR DESCRIPTION
**Pull request by rbberger**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed both new sha256.

**Comment by tldahlgren** [comment]
Will defer merge decision until at least one maintainer has a chance to comment.

File: var/spack/repos/builtin/packages/charliecloud/package.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Charliecloud(AutotoolsPackage):
     tags = ["e4s"]
 
     version("master", branch="master")
+    version("0.29", sha256="c89562e9dce4c10027434ad52eaca2140e2ba8667aa1ec9eadf789b4d7c1a6db")
```
Comment (j-ogas) [inline var/spack/repos/builtin/packages/charliecloud/package.py:20]: One thing to note is that 0.29 introduces a git build cache that requires git 2.28.1 or greater.

We should add something like:

```
depends_on("git@2.28.1:", type="run", when="@0.29:")
```

**Review by j-ogas — CHANGES_REQUESTED** [review]
See 0.29 git dependency. Aside from that, thanks!

File: var/spack/repos/builtin/packages/charliecloud/package.py (hunk 1 above)
Comment (rbberger) [reply var/spack/repos/builtin/packages/charliecloud/package.py:20]: done.

**Review by j-ogas — APPROVED** [review]
Looks good to me.

File: var/spack/repos/builtin/packages/charliecloud/package.py (hunk 1 above)
Comment (reidpr) [reply var/spack/repos/builtin/packages/charliecloud/package.py:20]: It doesn't need to be 2.28.1 exactly, and in fact if there's a choice, the newest available Git is likely the best one.

**Review by reidpr — COMMENTED** [review]
It doesn't need to be Git 2.28.1 exactly, and in fact if there's a choice, the newest available Git is likely the best one.

**Comment by j-ogas** [comment]
> It doesn't need to be Git 2.28.1 exactly, and in fact if there's a choice, the newest available Git is likely the best one.

I think the `git@2.28.1:` syntax, specifically the colon, sets 2.28.1 as the lower limit. That said, I'm not sure if it sets 2.28.1 as the default if not specified.

**Comment by rbberger** [comment]
@reidpr `2.28.1:` means 2.28.1 **or newer**.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**